### PR TITLE
feat: [CDS-50601]: Pipeline List / Pipeline Execution List Filter API update / Template default to Runtime Input

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

- Revert back to multi select dropdown in case of pipeline list filters and execution list filters with new API integrations
- Template -> Service Environments default to runtime values at all scopes
- Adding count `scopewise` of selected items in the scoped modals

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by vikrant-harness** [comment]
retrigger titlecheck

File: src/modules/10-common/components/EntityReference/EntityReference.tsx [hunk 1]
```diff
@@ -218,6 +218,19 @@ export function EntityReference<T extends ScopedObjectDTO>(props: EntityReferenc
   // used for multiselect
   const [selectedRecords, setSelectedRecords] = useState<ScopeAndIdentifier[]>(selectedRecordsFromProps ?? [])
 
+  const getCountOfSelectedRecordsInGivenScope = (scope: string | undefined): number => {
+    if (typeof scope !== 'string') {
```
Comment (AshwinBhatkal-Harness) [inline src/modules/10-common/components/EntityReference/EntityReference.tsx:222]: Let's extract this to a util. Can unit test easily

File: src/modules/70-pipeline/components/PipelineStudio/PipelineUtils.tsx [hunk 2]
```diff
@@ -13,3 +13,8 @@ export const isContextTypeNotStepGroupTemplate = (contextType?: string) =>
 
 export const isContextTypeStageOrStepGroupTemplate = (contextType?: string) =>
   contextType === PipelineContextType.StageTemplate || contextType === PipelineContextType.StepGroupTemplate
+
+export const isContextTypeTemplateType = (contextType?: string) =>
+  contextType === PipelineContextType.PipelineTemplate ||
+  contextType === PipelineContextType.StageTemplate ||
+  contextType === PipelineContextType.StepGroupTemplate
```
Comment (AshwinBhatkal-Harness) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineUtils.tsx:20]: Step group template condition will not be applicable in our case

Comment (vikrant-harness) [reply src/modules/70-pipeline/components/PipelineStudio/PipelineUtils.tsx:20]: Done

File: src/modules/10-common/components/EntityReference/EntityReference.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/10-common/components/EntityReference/EntityReference.tsx]: Done

**Review by AshwinBhatkal-Harness — DISMISSED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by vkbansal-harness — APPROVED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by udayvunnam — APPROVED** [review]

**Review by vardanbansal-harness — APPROVED** [review]

**Review by shaurya-harness — APPROVED** [review]

**Pull request merged**